### PR TITLE
Adds an app engine config directory chooser to the flex facet ui

### PR DESCRIPTION
**Pull request by etanshaul**
In support of #1373 

Breaking up #1373 into two PRs for reviewability:

1) (this pr) - adds a new field to the flex facet for setting an additional config directory - see screenshots below
2) (next pr) - consumes this field for deploying additional assets during flex deployment

UI changes:

valid state
image

invalid directory state:
image

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java [hunk 1]
```diff
@@ -166,7 +179,19 @@ private String getRuntimeText() throws MalformedYamlFileException {
    */
   private boolean isValidConfigurationFile(String path) {
     try {
-      if (!Files.isRegularFile(Paths.get(path))) {
+      if (StringUtils.isEmpty(path) || !Files.isRegularFile(Paths.get(path))) {
```
Comment (nkibler) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: For brevity, why not:

```
return !StringUtils.isEmpty(path) && Files.isRegularFile(Paths.get(path));
```

Then you don't need `return true;` at the bottom.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java [hunk 2]
```diff
@@ -166,7 +179,19 @@ private String getRuntimeText() throws MalformedYamlFileException {
...
+    } catch (InvalidPathException ipe) {
+      return false;
+    }
+
+    return true;
+  }
+
+  private boolean isValidConfigurationDirectory(String path) {
+    try {
+      if (StringUtils.isEmpty(path) || !Files.isDirectory(Paths.get(path))) {
```
Comment (nkibler) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: ditto

File: google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties [hunk 3]
```diff
@@ -81,12 +81,15 @@ appengine.flex.user.specified.artifact.label=Select a deployment archive:
 appengine.flex.dockerfile.location.label=Dockerfile path\:
 appengine.flex.version.placeholder.text=auto generated
 appengine.flex.facet.config.title=App Engine Flexible Configuration
+appengine.flex.facet.config.additional.config.title=Additional Configuration
 appengine.flex.facet.config.appyaml.error.before.text=The specified app.yaml is invalid or does not exist. Browse to an app.yaml file, or 
 appengine.flex.facet.config.appyaml.malformed.error=The specified app.yaml file is malformed. Browse to an app.yaml file, or 
 appengine.flex.facet.config.dockerfile.directory.error.before.text=The directory does not contain a Dockerfile. Browse to a directory with a Dockerfile, or 
+appengine.flex.facet.config.addition.config.directory.error=The specified directory does not exist
```
Comment (etanshaul) [inline google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: I think we've mostly been using periods. I'll add one here.

Comment (etanshaul) [inline google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: done

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java (hunk 1 above)
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java]: done

**Pull request merged**